src/extension/bazel_wrapper_commands.ts

```diff
@@ -36,12 +37,11 @@ async function bazelBuildTarget(adapter: IBazelCommandAdapter | undefined) {
     // If the command adapter was unspecified, it means this command is being
     // invoked via the command palatte. Provide quickpick build targets for
     // the user to choose from.
-    const quickPick = await vscode.window.showQuickPick(
-      queryQuickPickTargets({ query: "kind('.* rule', ...)" }),
-      {
-        canPickMany: false,
-      },
-    );
+    const quickPick = await showDynamicQuickPick({
+      query: "kind('.* rule', //...)",
```

Comment:
To me using `//...` as a placeholder here isn't that obvious. How about if this were instead a function, for example:

```
pattern => `kind('.* rule', ${pattern})`
```

Reply:
Great idea. I changed the interface to take both: A pattern and a builder function. I think this makes the code easier to understand.